src/rgw/rgw_op.cc

```diff
@@ -333,7 +335,7 @@ static int get_obj_head(struct req_state *s,
   std::unique_ptr<rgw::sal::RGWObject::ReadOp> read_op = obj->get_read_op(s->obj_ctx);
   obj->set_prefetch_data(s->obj_ctx);
 
-  int ret = read_op->prepare(s->yield);
+  int ret = read_op->prepare(s->yield, s);
```

Comment:
it's convenient to reuse `req_state` as the dpp here, but `RGWOp` is wrapping `req_state` as a prefix provider to print extra context

this (and several helper functions below) should take the `dpp` as a separate argument, which the RGWOp classes will pass in as `this`